### PR TITLE
Swap route settings.configure-secret-backend for nested edit and index route under secret.configuration

### PR DESCRIPTION
**Pull request by Monkeychip**
- [x] Enterprise test pass

### Description
This PR moves the route `vault.cluster.settings.confgure-secret-backend` to `vault.cluster.secrets.backend.configuration.edit`. It has also been renamed to "configuration/edit(js/hbs)". 

To keep the `vault.cluster.secrets.backend.configuration` route URL and behavior the same, I moved the code from `vault/cluster./secrets/backend/configuration` to `vault/cluster/secrets/backend/configuration/index`. This way I could  render the edit route without using {{outlet}}}. If there's another way to do this, I'm all ears.

**NO REDIRECT:** We intentionally opted to not redirect if a user navigated to the old route. We could have either caught the old url in the page-error component, or kept the old route around and transitioned in a beforeModel hook. However, modifying a root component like page-error or keeping a legacy route around for a url that was not likely to be bookmarked seemed like too much code debt.

**Why?** 
* The route FKA `configure-secret-backend` was responsible for configuring secret-engines, but was a child of `vault.cluster.settings` and not  `vault.cluster.secrets.backend`. This made it difficult to access secret-engine things. You can see [this struggle](https://github.com/hashicorp/vault/blob/main/ui/app/routes/vault/cluster/settings/configure-secret-backend.js#L17) in how the route has to query all the secret engines and search for the type to see if the route should throw an error or proceed. However, if it was a child route of `vault.cluster.secrets.backend` it would already have access to the secret-engine model and would not need to query all them and then search through. 
* IMO, it makes sense here (the [old path](https://github.com/hashicorp/vault/blob/main/ui/app/router.js#L53) override also seems to agree). I'm actually not sure why it was under settings at all other than `mount-secret-backend` is under settings and they're related. Though, I personally don't think `mount-secret-backend` should be under settings either  🥫🪱.  
* This route is currently only used for two secret-engines: aws and ssh. It will be used for more and only get more complicated.

**Notes** 
* I ONLY moved files in this PR. There is sooo much more I could do (glimmerize, refactor methods, address current bugs). But because this is a significant change (anything routing always is), I wanted to make it clear what was actually happening. Those TODOs are coming, but not in this PR.
* TLDR: I don't have to merge this. I can continue my WIF work without these changes, but it started to feel like I was pilling fresh food onto a dirty plate. I don't have the time to clean the whole kitchen, but maybe tidy up a bit?

### CODE REVIEW

File: ui/app/routes/vault/cluster/secrets/backend/configuration/edit.js
Comment (Monkeychip) [inline ui/app/routes/vault/cluster/secrets/backend/configuration/edit.js:1]: I tried model(params) thinking params.backend would be a thing. But I apparently don't understand [this comment correctly](https://api.emberjs.com/ember/5.10/classes/Route/methods/model?anchor=model). Anyone know why, even though `backend` is a dynamic segment on the URL, I can't access it via the model params?

Comment (hashishaw) [reply ui/app/routes/vault/cluster/secrets/backend/configuration/edit.js:1]: Oh, it's because `backend` is a param on a parent route (`secrets.backend`) instead of the current one. So you could do `this.paramsFor('vault.cluster.secrets.backend')` to get the backend path

File: ui/app/components/not-found.js [hunk 1]
```diff
@@ -20,7 +20,16 @@ import Component from '@glimmer/component';
 export default class NotFound extends Component {
   @service router;
 
-  get path() {
-    return this.router.currentURL;
+  constructor() {
+    super(...arguments);
+    const { currentURL } = this.router;
+    if (currentURL.startsWith('/vault/settings/secrets/configure/')) {
```
Comment (hashishaw) [inline ui/app/components/not-found.js:26]: My only hesitation with this is if a user is trying to get to the configure page of a backend that doesn't exist, we might get into an infinite loop of redirects. I might prefer that we keep the old route for now, show a flash message on redirect alerting the user that the route will be removed in future versions, and then have a deprecation schedule for the route. 
EDIT: I checked this locally, and it looks like attempting to access a backend that doesn't exist errors with a 403 instead of a 404 so the not-found template doesn't render, which is good. May still be worth evaluating our route deprecation approach generally because this has come up before and we didn't really have a cohesive plan

File: ui/app/router.js [hunk 2]
```diff
@@ -172,7 +168,10 @@ Router.map(function () {
           this.mount('ldap');
           this.mount('pki');
           this.route('index', { path: '/' });
-          this.route('configuration');
+          this.route('configuration', function () {
+            // only CONFIGURABLE_SECRET_ENGINES can be configured and access the edit route
+            this.route('edit');
```
Comment (hashishaw) [inline ui/app/router.js:173]: I double checked locally but it looks like we hadn't been using the `section` route under `configure-secret-backend` ✅

File: ui/tests/acceptance/secrets/backend/aws/aws-configuration-test.js [hunk 3]
```diff
@@ -63,6 +63,21 @@ module('Acceptance | aws | configuration', function (hooks) {
     await runCmd(`delete sys/mounts/${path}`);
   });
 
+  test('it should redirect if old url is entered', async function (assert) {
+    const path = `aws-${this.uid}`;
+    await enablePage.enable('aws', path);
+    await click(SES.configTab);
+    await visit(`/vault/settings/secrets/configure/${path}`);
```
Comment (hashishaw) [inline ui/tests/acceptance/secrets/backend/aws/aws-configuration-test.js:70]: Since there used to be another `/section` piece of this URL path, I wonder if we should also have a test and case handling for a url like `/vault/settings/secrets/configure/${path}/leases` in case someone has a really old URL bookmarked

**Review by hashishaw — COMMENTED** [review]
Nice work, this change makes a lot of sense! I think there are some routing change process things to consider before we go ahead and ship this, so I'm holding off approval for now

File: ui/app/router.js (hunk 2 above)
Comment (Monkeychip) [reply ui/app/router.js:173]: That one threw me. I searched and searched, and came to the same conclusion.

File: ui/tests/acceptance/secrets/backend/aws/aws-configuration-test.js (hunk 3 above)
Comment (Monkeychip) [reply ui/tests/acceptance/secrets/backend/aws/aws-configuration-test.js:70]: Good idea!

**Review by hashishaw — APPROVED** [review]
Beautiful! 🚢
